src/software/geom/spline_test.cpp

```diff
@@ -0,0 +1,99 @@
+#include "software/geom/spline.h"
+
+#include <gtest/gtest.h>
+
+TEST(TestSpline, test_spline_points_constructor)
+{
+    std::vector<Point> points({Point(4, 18), Point(3, 0), Point(-7, -1)});
+    Spline s(points);
+    EXPECT_EQ(s.size(), points.size());
+    std::vector<Point> spline_points = s.getKnots();
+    EXPECT_EQ(spline_points, points);
+    EXPECT_EQ(s.valueAt(0.0), points[0]);
+    EXPECT_EQ(s.startPoint(), points[0]);
+    EXPECT_EQ(s.valueAt(0.1), Point(3.8, 14.4));
+    EXPECT_EQ(s.valueAt(0.5), points[1]);
+    EXPECT_EQ(s.valueAt(0.6), Point(1, -0.2));
+    EXPECT_EQ(s.valueAt(1.0), points[2]);
+    EXPECT_EQ(s.endPoint(), points[2]);
```

Comment:
We're testing a bunch of stuff here, could we use the minimum set of getters for testing the constructor, and have seperate tests for things like `startPoint`, `endPoint`, `valueAt`?

In general we should really try to avoid large monolithic tests like this, as they're harder to follow, more vague about what they're actually testing, and less indicative of where problems actually are when they fail.

Reply:
Yeah my bad. It was manageable until I expanded the class with new functions and added new lines to test those

Reply:
done